File: hw_1.js
function base62toBase10(numToConvert){
    var base62Lookup = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
    var convertedNumber = 0;
    var base = 62;

    // move backwards through the base62 number string and
    // use a varible starting at 0 to compute the power
    // of 62 at that place
    for (var i = numToConvert.length - 1, j = 0; i >= 0; i--, j++){
        var currentLetter = numToConvert[i];
        var decimalValue = base62Lookup.indexOf(currentLetter);
        convertedNumber += decimalValue * base ** j;
    }

    return convertedNumber;
}

// 18327995462734720000
console.log(base62toBase10("LpuPe81bc2w"));

File: hw_1.py
import math

def base62toBase10(numToConvert):
    base62Lookup = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
    convertedNumber = 0
    currentBase = 0

    for i in range(len(numToConvert)-1, -1, -1):
        currentLetterValue = base62Lookup.find(numToConvert[i])
        convertedNumber += currentLetterValue * 62 ** currentBase
        # convertedNumber = convertedNumber * 62 + currentLetterValue
        currentBase += 1

    return convertedNumber

print(base62toBase10("LpuPe81bc2w"))